src/averageAges.js

```diff
@@ -56,6 +73,18 @@ function calculateWomenAverageAge(people, withChildren) {
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
   // write code here
+  return people
+    .filter(child => onlyWithSon ? child.sex === 'm' : true)
+    .filter(child => getMother(people, child))
+    .reduce((average, child, index, children) => {
+      const mother = getMother(people, child);
+
+      return average + (child.born - mother.born) / children.length;
+    }, 0);
+}
+
+function getMother(people, child) {
+  return people.find(mother => mother.name === child.mother);
```

Comment:
you don't need an actual object - but just confirm it exists - use `.some` method